state-transition/core/metrics.go

```diff
@@ -75,3 +57,33 @@ func (s *stateProcessorMetrics) gaugeBlockGasUsed(
 		blockNumberStr,
 	)
 }
+
+func (s *stateProcessorMetrics) gaugePendingPartialWithdrawalsCount(count int) {
+	s.sink.SetGauge("beacon_kit.state.pending_partial_withdrawals_count", int64(count))
+}
+
+func (s *stateProcessorMetrics) gaugeTimestamps(
+	payloadTimestamp uint64,
+	consensusTimestamp uint64,
+) {
+	// the diff can be positive or negative depending on whether the payload
+	// timestamp is ahead or behind the consensus timestamp
+	diff := int64(payloadTimestamp) - int64(consensusTimestamp) // #nosec G115
+	s.sink.SetGauge("beacon_kit.state.payload_consensus_timestamp_diff", diff)
+}
+
+func (s *stateProcessorMetrics) incrementDepositStakeLost() {
+	s.sink.IncrementCounter("beacon_kit.state.deposit_stake_lost")
+}
+
+func (s *stateProcessorMetrics) incrementPartialWithdrawalRequestDropped() {
+	s.sink.IncrementCounter("beacon_kit.state.partial_withdrawal_request_dropped")
+}
+
+func (s *stateProcessorMetrics) incrementPartialWithdrawalFailed() {
```

Comment:
how about `incrementInvalidPartialWithdrawal`s?
I feel if people reads failed they would assume the chain is malfunctioning instead of understanding that there is something wrong with the withdrawal?